src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+    const signedUserId = await getUserId();
+    const anonUserId = await getAnonymousUserId();
+    const token = TokenStorage.get(ITokenEnum.token);
+    const anonymousToken = TokenStorage.get(ITokenEnum.anonymousToken);
+
+    setProfileAtom({
+      anonProfileId: anonUserId,
+      signedProfileId: signedUserId,
+      token,
+      anonymousToken
+    });
+  };
+  let isOpenNotification = false;
+
+  const helperNavigationResetWithData = (screenData) => {
+    setTimeout(() => {
+      if (!isOpenNotification) {
+        isOpenNotification = true;
+        const routes = [
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen: 'HomeTabs',
+              params: {
+                screen: 'SignedChannelList'
+              }
+            }
+          }
+        ];
+
+        if (screenData)
+          routes.push({
+            name: 'AuthenticatedStack',
+            params: {
+              ...screenData
+            }
+          });
+
+        navigation.reset({
+          index: screenData ? 2 : 1,
+          routes
+        });
+      }
+    }, 500);
+  };
+
+  const __handleNotification = async (notification) => {
+    if (notification.data.type === 'feed' || notification.data.type === 'reaction') {
+      navigation.navigate('PostDetailPage', {
+        feedId: notification.data.feed_id,
+        is_from_pn: true
+      });
+    }
+    if (notification.data.type === 'follow_user') {
+      navigation.navigate('OtherProfile', {
+        data: {
+          user_id: notification.data.user_id,
+          other_id: notification.data.user_id_follower,
+          username: notification.data.username_follower
+        }
+      });
+    }
+    if (notification.data.type === 'message.new') {
+      if (notification?.data?.receiver_id === signedProfileId) {
+        const channel = new ChannelList({
+          id: notification?.data?.channel_id,
+          channelType: 'PM'
+        });
+        setIsLoadingFetchingChannelDetail(true);
+        try {
+          await fetchChannelDetail(channel);
+          const selectedChannel = await ChannelList.getSchemaById(
+            localDb,
+            notification?.data?.channel_id
+          );
+          setSelectedChannel(selectedChannel);
+          helperNavigationResetWithData({
+            screen: 'SignedChatScreen'
+          });
+        } catch (e) {
+          console.log('error', e);
+        } finally {
+          setIsLoadingFetchingChannelDetail(false);
+        }
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    __createChannel();
+    __updateProfileAtomId();
+
+    messaging().setBackgroundMessageHandler(async (remoteMessage) => {
+      __handlePushNotif(remoteMessage);
+    });
+
+    const unsubscribe = messaging().onMessage((remoteMessage) => {
+      // eslint-disable-next-line no-unused-expressions
+      __handlePushNotif(remoteMessage);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this useEffect hook, you're setting up the push notification channel and updating the profile atom when the component mounts. You're also setting up a background message handler for Firebase messaging. This is a good use of the useEffect hook to perform setup tasks.



<!-- This is an auto-generated comment by OSS CodeRabbit -->